File: VB/MySolution.Module/BusinessObjects/Task.vb
Imports DevExpress.ExpressApp.DC
Imports DevExpress.Persistent.Base
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

<DefaultClassOptions, ImageName("BO_Task"), MetadataType(GetType(TaskMetadata))> _
    Partial Public Class Task
End Class
Public Class TaskMetadata
    <Browsable(False)> _
    Public Property Id() As Int32
    <FieldSize(FieldSizeAttribute.Unlimited)> _
    Public Property Description() As String
End Class

File: CS/MySolution.Module/BusinessObjects/Task.cs
using DevExpress.ExpressApp.DC;
using DevExpress.Persistent.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MySolution.Module {
    [DefaultClassOptions, ImageName("BO_Task")]
    [MetadataType(typeof(TaskMetadata))]
    public partial class Task {
    }
    public class TaskMetadata {
        [Browsable(false)]
        public Int32 Id { get; set; }
        [FieldSize(FieldSizeAttribute.Unlimited)]
        public String Description { get; set; }
    }

}
